apps/mark-scan/frontend/src/pages/diagnostics_screen.tsx

```diff
@@ -96,46 +23,66 @@ export function DiagnosticsScreen({
   onBackButtonPress,
 }: DiagnosticsScreenProps): JSX.Element {
   const batteryQuery = systemCallApi.getBatteryInfo.useQuery();
-  const battery = batteryQuery.data ?? undefined;
+  const mostRecentAccessibleControllerDiagnosticQuery =
+    getMostRecentAccessibleControllerDiagnostic.useQuery();
+  const diskSpaceQuery = getApplicationDiskSpaceSummary.useQuery();
 
-  const [
-    accessibleControllerDiagnosticResults,
-    setAccessibleControllerDiagnosticResults,
-  ] = useState<AccessibleControllerDiagnosticResults>();
   const history = useHistory();
 
+  if (
+    !batteryQuery.isSuccess ||
+    !mostRecentAccessibleControllerDiagnosticQuery.isSuccess ||
+    !diskSpaceQuery.isSuccess
+  ) {
```

Comment:
If an error is thrown, it will hit the app error boundary. There is also a react query error state, but we typically don't plan for that because we don't see it in our off-line environment.